### PR TITLE
Feature/accelerators

### PR DESCRIPTION
**Pull request by roxanajula**
The **Apple Gestures** and **Custom Gesture Recognizer** blog posts are part of a 3 blog post series on App Accelerators, the other blog post : **App Accelerators** will be launched after those 2 as I am referencing them in it and I will need the links for them.

### CODE REVIEW

File: source/_posts/2017-12-08-Custom-Gesture-Recognizers.md [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+
+There are 2 types of custom gesture recognisers you can define yourself:
+
+* **Discrete** - when your gesture involves a specific pattern of events. Examples: recognising a checkmark, a circle or a chevron (see example below). Those are easier to implement than the continuous ones because of the number of states they can have. 
+* **Continuous** - when your gesture does not follow a predefined pattern, when you want to gather user input and there are no failure cases. Example: recognising a user’s drawings on a screen. This usually requires a bit more effort to implement.
+
+## Discrete Gesture Recognizer example
+For learning purposes we chose a simple example of recognising a right chevron and open the photo library. In real apps, you might want to look into other standard gestures for an accelerator for opening the photo library.
+
+<Add photo with gesture image>
```
Comment (selynodes) [inline source/_posts/2017-12-08-Custom-Gesture-Recognizers.md]: We should add photo?

File: source/_posts/2017-12-08-Custom-Gesture-Recognizers.md [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+
+### State machine
+Before we get into it we should understand the state machine. This will be a recognizer for a discrete gesture (a specific pattern) so we will look into how we manage transitions for this case only. If you want to learn more about the state machine for continuous gestures you can check out [Apple’s documentation](https://developer.apple.com/documentation/uikit/touches_presses_and_gestures/implementing_a_custom_gesture_recognizer/implementing_a_continuous_gesture_recognizer).
```
Comment (selynodes) [inline source/_posts/2017-12-08-Custom-Gesture-Recognizers.md]: change `Apple’s`to `Apple's`

File: source/_posts/2017-12-08-Custom-Gesture-Recognizers.md [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+
+A gesture recognizer always starts in the **possible** state, that tells us that we are ready to process events. Based on the user’s input we can end up in a **failed** (unsuccessful) state or **ended** (successful) state. You have to check if the user is following your pattern and at any deviation from that you should set the state to failed. If the user did indeed followed the expected pattern, UIKit will call the action method associated to the target object.
```
Comment (selynodes) [inline source/_posts/2017-12-08-Custom-Gesture-Recognizers.md]: change `user’s` to `user's`

Comment (selynodes) [reply source/_posts/2017-12-08-Custom-Gesture-Recognizers.md]: > user did indeed followed the expected pattern

Shouldn't this be `...user did indeed follow the expected pattern...`

File: source/_posts/2017-12-08-Custom-Gesture-Recognizers.md [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+
+We also have the **cancelled** state that occurs when the current event is interrupted by a system event like receiving a phone call. You can also customise the cancel situations that can happen in your app. This is to make sure that we don’t perform tasks that the user didn’t intend.
```
Comment (selynodes) [inline source/_posts/2017-12-08-Custom-Gesture-Recognizers.md]: `don’t` should be change to `do not` or `don't`

Comment (selynodes) [reply source/_posts/2017-12-08-Custom-Gesture-Recognizers.md]: same goes for `didn’t`----> `didn't`

File: source/_posts/2017-12-08-Custom-Gesture-Recognizers.md [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+
+It is also important to implement a **reset()** method that will make sure we are ready to process new event sequences by returning to starting values.
+
+<p align=center><img width=50%; src="https://docs-assets.developer.apple.com/published/eedc7ffb40/922db1b2-9860-4618-8226-699b360701b3.png"/></p>
+
+*[Image source](https://developer.apple.com/documentation/uikit/touches_presses_and_gestures/implementing_a_custom_gesture_recognizer/about_the_gesture_recognizer_state_machine)*
+
+### Custom gesture recognizer
+The more conditions you have for a gesture to be successful the more precise it will be. For this example we expect that the user will use only one finger for the gesture, makes a down stroke from left to right followed by a down stroke from right to left. Those conditions are leaving open possibilities for edge cases but for the sake of this example we won’t look too much into that.
```
Comment (selynodes) [inline source/_posts/2017-12-08-Custom-Gesture-Recognizers.md]: `wont´t` to `won't`

File: source/_posts/2017-12-08-Custom-Gesture-Recognizers.md [hunk 6]
```diff
@@ -0,0 +1,192 @@
+---
+title:  Custom Gesture Recognizers
+date: 2017-12-08 13:00:00
```
Comment (selynodes) [inline source/_posts/2017-12-08-Custom-Gesture-Recognizers.md]: Remember to bump the date!

File: source/_posts/2017-12-08-UIGestures.md [hunk 7]
```diff
@@ -0,0 +1,95 @@
+---
+title:  UIGestures
+date: 2017-12-08 13:00:00
```
Comment (selynodes) [inline source/_posts/2017-12-08-UIGestures.md]: Remember to bump the date

File: source/_posts/2017-12-08-UIGestures.md [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+tags: ios,accelerators,gestures
+authorIds:
+- roju
+categories:
+- iOS
+---
+
+Gestures are the magic behind those accelerators that can make your app more user friendly. We will show now a few examples on how to implement them in your code.
```
Comment (selynodes) [inline source/_posts/2017-12-08-UIGestures.md]: > We will show now a few examples...

Maybe change order of `show` and `now` 

> We will now show a few examples...

File: source/_posts/2017-12-08-UIGestures.md [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+
+## [Swipe Gestures](https://developer.apple.com/documentation/uikit/uiswipegesturerecognizer)
+
+You can have many accelerators build with the swipe gesture as it is discrete and you can have many combinations as you can select the direction of the swipe and the number of touches required. Even if we have the possibility to set up a high number of touches, a maximum of 3 is still user friendly. You should also make sure you keep some sort of consistency when defining the gestures, if swipe right with 2 fingers goes to the previous conversation in a messaging app, swiping left with 2 fingers should probably go to the next conversation.
+
+### Using the storyboard
+
+{% vimeo 244338896 %}
```
Comment (selynodes) [inline source/_posts/2017-12-08-UIGestures.md]: Is there someway to make these videos display larger?
screen shot 2017-12-18 at 14 37 52

File: source/_posts/2017-12-08-UIGestures.md [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+
+1. Search for the swipe gesture in the objects library
+2. Drag it into your view controller
+3. Set the view of your choice as a delegate to the swipe gesture recognizer (by pressing “Ctrl” on the keyboard and click-drag from the view to the swipe gesture recognizer and selecting Outlet Collections - `gestureRecognizers` )
+4. Setup your gesture recogniser by selecting the direction and the number of touches required
+
+{% vimeo 244338890 %}
```
Comment (selynodes) [inline source/_posts/2017-12-08-UIGestures.md]: Is there someway to make these videos display larger?

**Comment by roxanajula** [comment]
@chriscombs do you know how to make the vimeo videos bigger?

**Comment by chriscombs** [comment]
I'm not really sure how the media works, it always renders differently on the actual blog than on github

**Comment by roxanajula** [comment]
@chriscombs then should we launch it as is now?

**Comment by mariusc** [comment]
Don't know if things have changed, but back in the days, you could use `npm run write` to run it locally. https://github.com/nodesagency/blog/wiki/D-01.-Cloning-and-installing#running-the-site-locally. Of course, that assumes you have the version of node, npm, hexo and gulp that it's expecting. Depending on how much you want to play with those, it could be worth trying to set it up locally. 

Another choice was to deploy it to staging and see how it looks (the url of the staging website was pinned in the tech blog channel in slack)

**Comment by roxanajula** [comment]
@mariusc Thanks!Yes I was trying to do that today but i ran into some node npm hexo problems. Hopefully I will fix them soon and I will be able to stat testing locally.

**Pull request merged**